plugins/topology/src/types/types.ts

```diff
@@ -0,0 +1,133 @@
+export type K8sModel = {
+  abbr: string;
+  kind: string;
+  label: string;
+  labelPlural: string;
+  plural: string;
+  propagationPolicy?: 'Foreground' | 'Background';
+  id?: string;
+  apiVersion: string;
+  apiGroup?: string;
+  namespaced?: boolean;
+  color?: string;
+};
+
+type OwnerReference = {
+  name: string;
+  kind: string;
+  uid: string;
+  apiVersion: string;
+  controller?: boolean;
+  blockOwnerDeletion?: boolean;
+};
+
+type ObjectMetadata = {
+  annotations?: { [key: string]: string };
+  clusterName?: string;
+  creationTimestamp?: string;
+  deletionGracePeriodSeconds?: number;
+  deletionTimestamp?: string;
+  finalizers?: string[];
+  generateName?: string;
+  generation?: number;
+  labels?: { [key: string]: string };
+  managedFields?: any[];
+  name?: string;
+  namespace?: string;
+  ownerReferences?: OwnerReference[];
+  resourceVersion?: string;
+  uid?: string;
+};
+
+// Properties common to (almost) all Kubernetes resources.
+type K8sResourceCommon = {
+  apiVersion?: string;
+  kind?: string;
+  metadata?: ObjectMetadata;
+};
+
+// Generic, unknown kind. Avoid when possible since it allows any key in spec
+// or status, weakening type checking.
+export type K8sResourceKind = K8sResourceCommon & {
```

Comment:
We're trying to slowly remove this from OpenShift console, so would be good to avoid using it here if we can. It's the `[key: string]: any` which makes the type checking much weaker even if you extend this type because you don't catch any typos in property names.

Reply:
so shall I add strict types for supported resources over generic ones like K8sDeploymentKind, K8sPodKind, etc?

Reply:
i checked backstage-k8s plugin  and it fetches native types from https://www.npmjs.com/package/@kubernetes/client-node , have used the same to be more strict

https://github.com/backstage/backstage/blob/master/plugins/kubernetes/package.json#L43
https://github.com/backstage/backstage/blob/master/plugins/kubernetes/src/components/CronJobsAccordions/CronJobsAccordions.tsx#L25